Description: Add the ares_get_config() routine used by Unreal IRCd.
Origin: the Unreal IRCd distribution
Forwarded: no
Last-Update: 2010-03-22

--- ares.h.orig
+++ ares.h
@@ -261,6 +261,14 @@
   int nsort;
 };
 
+/** Public available config (readonly) interface for ares_get_config(). */
+struct ares_config_info {
+	int timeout;
+	int tries;
+	int numservers;
+	char **servers;
+};
+
 struct hostent;
 struct timeval;
 struct sockaddr;
@@ -487,6 +495,8 @@
 
 CARES_EXTERN const char *ares_strerror(int code);
 
+CARES_EXTERN int ares_get_config(struct ares_config_info *d, ares_channel c);
+
 #ifdef  __cplusplus
 }
 #endif
--- ares_init.c.orig
+++ ares_init.c
@@ -1498,6 +1498,26 @@
 }
 #endif /* !WIN32 && !WATT32 */
 
+int ares_get_config(struct ares_config_info *d, ares_channel c)
+{
+	int i;
+	char *p;
+
+	memset(d, 0, sizeof(struct ares_config_info));
+
+	d->timeout = c->timeout;
+	d->tries = c->tries;
+	d->numservers = c->nservers;
+	d->servers = calloc(sizeof(char *), c->nservers);
+	for (i = 0; i < c->nservers; i++)
+	{
+		p = inet_ntoa(c->servers[i].addr);
+		d->servers[i] = p ? strdup(p) : NULL;
+	}
+
+	return ARES_SUCCESS;
+}
+
 /* initialize an rc4 key. If possible a cryptographically secure random key
    is generated using a suitable function (for example win32's RtlGenRandom as
    described in
